src/tracing/errors.ts

```diff
@@ -9,32 +9,58 @@ declare global {
   }
 }
 
-/** Identifies ethers request errors (as thrown by {@type import(@ethersproject/web).fetchJson}). */
+// Identifies ethers request errors (as thrown by {@type import(@ethersproject/web).fetchJson}).
 function isEthersRequestError(error: Error): error is Error & { requestBody: string } {
   return 'requestBody' in error && typeof (error as unknown as Record<'requestBody', unknown>).requestBody === 'string'
 }
 
-export function beforeSend(event: ErrorEvent, hint: EventHint) {
-  // Since the interface currently uses HashRouter, URLs will have a # before the path.
-  // This leads to issues when we send the URL into Sentry, as the path gets parsed as a "fragment".
-  // Instead, this logic removes the # part of the URL.
-  // See https://romain-clement.net/articles/sentry-url-fragments/#url-fragments
+// Since the interface currently uses HashRouter, URLs will have a # before the path.
+// This leads to issues when we send the URL into Sentry, as the path gets parsed as a "fragment".
+// Instead, this logic removes the # part of the URL.
+// See https://romain-clement.net/articles/sentry-url-fragments/#url-fragments
+function updateRequestUrl(event: ErrorEvent) {
   if (event.request?.url) {
     event.request.url = event.request.url.replace('/#', '')
   }
+}
+
+// If a request fails due to a chunk error, this looks for that asset in the performance entries.
+// If found, it adds a tag to the event with the response status of the chunk request.
+function addChunkResponseStatusTag(event: ErrorEvent, hint: EventHint) {
+  const error = hint.originalException
+  if (error instanceof Error) {
+    let asset: string | undefined
+    if (error.message.match(/Loading chunk \d+ failed\. \(([a-zA-Z]+): .+\.chunk\.js\)/)) {
+      asset = error.message.match(/https?:\/\/.+?\.chunk\.js/)?.[0]
+    }
+
+    if (error.message.match(/Loading CSS chunk \d+ failed\. \(.+\.chunk\.css\)/)) {
+      const relativePath = error.message.match(/\/static\/css\/.*\.chunk\.css/)?.[0]
+      asset = `${window.origin}${relativePath}`
+    }
+
+    if (asset) {
+      const status = getChunkResponseStatus(asset)
+      if (status) {
+        if (!event.tags) {
+          event.tags = {}
+        }
+        event.tags.chunkResponseStatus = status
+      }
+    }
+  }
+}
 
+export function beforeSend(event: ErrorEvent, hint: EventHint) {
+  updateRequestUrl(event)
+  addChunkResponseStatusTag(event, hint)
   return filterKnownErrors(event, hint)
 }
```

Comment:
nit: put this at the top of the file - everything else is in scope due to hoisting. You already use hoisting for `getChunkResponseStatus`, so moving the exported bit to the very top (or the very bottom, without any hoisting) improves readability instead of leaving it in the middle.

Reply:
should we add a lint for this?